pylutron_caseta/smartbridge.py

```diff
@@ -42,60 +41,116 @@ def __init__(self, hostname=None):
             self.get_value(_id)
         self._subscribers = {}
 
-    def add_subscriber(self, device_id, _callback):
-        """Add a listener to be notified of state changes."""
-        self._subscribers[device_id] = _callback
+    def add_subscriber(self, device_id, callback_):
+        """
+        Add a listener to be notified of state changes.
+
+        :param device_id: device id, e.g. 5
+        :param callback_: callback to invoke
+        """
+        self._subscribers[device_id] = callback_
 
     def get_devices(self):
-        """Will return all known devices connected to the Smartbridge."""
+        """Will return all known devices connected to the Smart Bridge."""
         return self.devices
 
-    def get_devices_by_type(self, _type):
-        """Will return all devices of a given type."""
+    def get_devices_by_domain(self, domain):
+        """
+        Return a list of devices for the given domain.
+
+        :param domain: one of 'light', 'switch', 'cover' or 'sensor'
+        :returns list of zero or more of the devices
+        """
         devs = []
+
+        # return immediately if not a supported domain
+        if domain not in _LEAP_DEVICE_TYPES:
+            return devs
+
+        # loop over all devices and check their type
         for device_id in self.devices:
-            if self.devices[device_id]['type'] == _type:
+            if self.devices[device_id]['type'] in _LEAP_DEVICE_TYPES[domain]:
                 devs.append(self.devices[device_id])
         return devs
 
-    def get_devices_by_types(self, _types):
-        """Will return all devices of for a list of given types."""
+    def get_devices_by_type(self, type_):
+        """
+        Will return all devices of a given device type.
+
+        :param type_: LEAP device type, e.g. WallSwitch
+        """
         devs = []
         for device_id in self.devices:
-            if self.devices[device_id]['type'] in _types:
+            if self.devices[device_id]['type'] == type_:
+                devs.append(self.devices[device_id])
+        return devs
+
+    def get_devices_by_types(self, types):
```

Comment:
We should mark this is deprecated in favor of the get_by_domain().  We can remove this in a future version.

Reply:
@gurumitts Do you consider this as a standalone Caseta library? The function could be used for various reasons even if it is not from Home Assistant. For example, I could get all the 3 button Picos if I know what I'm looking for.

That said, if the primary use of the library is assumed to be Home Assistant (and indeed the 'light', 'cover', 'switch', etc. comes from HA) and you want to, I can mark it deprecated. How does one do that in Python? Do I add a logger warning statement? Let me know which you prefer.

Reply:
Ahh... good point.  Lets leave it alone for now.  Yes it is meant to me a stand alone lib.  I am merging this now.